### PR TITLE
Fixes and improvements for crud manager and generic list

### PR DESCRIPTION
**Pull request by tomaskikutis**
REQUIRED BY: https://github.com/superdesk/superdesk-planning/pull/1230

### CODE REVIEW

File: scripts/core/helpers/CrudManager.tsx [hunk 1]
```diff
@@ -161,24 +133,12 @@ export const dataApi: IDataApi = {
         const queryString = '?' + Object.keys(query).map((key) =>
```
Comment (petrjasek) [inline scripts/core/helpers/CrudManager.tsx]: imo would make sense to call `encodeURI` on that

File: scripts/core/helpers/CrudManager.tsx [hunk 2]
```diff
@@ -161,24 +133,12 @@ export const dataApi: IDataApi = {
...
             `${key}=${isObject(query[key]) ? JSON.stringify(query[key]) : query[key]}`).join('&');
```
Comment (petrjasek) [inline scripts/core/helpers/CrudManager.tsx]: or rather `encodeURIComponent(query[key])`

File: .eslintrc.js [hunk 3]
```diff
@@ -1,6 +1,35 @@
 const sharedConfigs = require('superdesk-code-style');
 
 module.exports = Object.assign({}, sharedConfigs, {
+    rules: Object.assign(sharedConfigs.rules, {
```
Comment (pablopunk) [inline .eslintrc.js]: Should include/discuss all this rules in https://github.com/superdesk/superdesk-client-code-style?

File: scripts/core/helpers/network.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+function httpRequestBase(options: IHttpRequestOptions): Promise<Response> {
+    const {method, url, payload, headers} = options;
+
+    return fetch(url, {
+        method,
+        headers: headers || {},
+        mode: 'cors',
+        body: typeof payload === 'undefined' ? undefined : JSON.stringify(payload),
```
Comment (pablopunk) [inline scripts/core/helpers/network.tsx]: I think the ternary is unnecessary here, what's the difference between this and `body: JSON.stringify(payload)`?

Comment (tomaskikutis) [reply scripts/core/helpers/network.tsx]: You are right, there's none. I thought `JSON.stringify` returns an `'undefined'` string as it does with `null` which turns out isn't true.

File: .eslintrc.js (hunk 3 above)
Comment (tomaskikutis) [reply .eslintrc.js]: I want to finish ansa stuff before going on holiday next week, but you can do it without me.

**Pull request merged**